### DISCUSSION TITLE
記事のタグ欄にgppやcppと入力したら、g++,c++に変換されてしまいます。

### DISCUSSION THREAD

**Discussion by fygar256** [increments/qiita-discussions#918] [要望・アイデア・不具合]
gppはプリプロセッサのことで、g++はGNUのc++コンパイラコレクションです。
違いますのに、どうしてでしょうか。うちだけでしょうか？

**Comment by kimkim0814** [comment]
@fygar256
お問合せありがとうございます。
ご不便をおかけして申し訳ありません。

タグが変換される理由は、Qiitaのタグ処理における名寄せのルールによるものです。
具体的には、以下のような変換規則が適用されています。

- cpp は C++ に変換され、URLは cpp になります。
- gpp は g++ に変換され、URLは gpp になります。

これは、タグの一貫性を保つために行われています。例えば、cpp や csharp のようなタグは、
名寄せ処理によって C++ や C# として統一されます。

現状、回避策としては、gnu-preprocessor のように具体的な名前に変更することで、変換を避けることができます。

`gpp` と `g++` は異なるものであり、現行の名寄せルールが適切でないと感じられる点については、貴重なご意見として受け取らせていただき、ルールの改訂も検討いたします。

どうぞよろしくお願いいたします。

**Reply by fygar256** [reply]
どうもご回答ありがとうございます。

**Comment by fygar256** [comment]
close
